src/RecoveryServices/RecoveryServices.Backup/Cmdlets/ProtectionPolicy/SetAzureRmRecoveryServicesBackupProtectionPolicy.cs

```diff
@@ -151,6 +152,27 @@ public override void ExecuteCmdlet()
                     throw new ArgumentException(string.Format(Resources.PolicyNotFoundException,
                         Policy.Name));
                 }
+                // Validate the setting of changing of backup tier here is workload is afs
+                if (Policy.WorkloadType == WorkloadType.AzureFiles)
```

Comment:
P1: can we move this workload specific validation to fileshare provider

Reply:
@hiaga Is it possible to use ShouldContinue()/ShouldProcess() outside of cmdlet execution block? I tried move the logic to fileshare provider but I am receiving an CS0120: An object reference is required for the non-static field, method, or property 'Cmdlet.ShouldContinue(string, string) error. I am unsure how can reference the cmdlet instance from within the provider. Additionally I cannot find any examples in codebase of ShouldContinue being used in workload providers.